opentracing/src/main/java/opentracing/Tracer.java

```diff
@@ -31,7 +36,9 @@
    *                   .setTag("lucky_number", 42);
    * }</pre>
    */
-  Span startTrace(String operationName);
+  Span startSpan(String operationName, Optional<Span> parent);
+
+  <T> Injector<T> getInjector(Class<T> carrierType);
```

Comment:
Maybe I'm not reading this right, but are you basically suggesting going towards https://github.com/opentracing/opentracing-go/pull/48#issuecomment-181047181? I.e. (this is copy-pasta from there):

```go
type Tracer interface {
  // Join attempts to obtain a Span from the given interface.
  Join (interface{}) (Span, error)
}

type Span interface {
  // Inject attempts to store the Span's information in the given interface.
  Propagate(interface{}) error
}

t := something.NewTracer()
sp := t.StartTrace("foo")
var req http.Request
// This will add the trace information to the request's headers.
if err := t.Propagate(&req); err != nil {
  return err
}

// Receiving end uses the Tracer to join a new Span onto the one
// encoded in the Request.
if sp, err := t.Join(&req); err != nil {
  // ...
}
```